### PR TITLE
Upgrade to Symfony v5.4 minimum

### LINKED ISSUE #263: [RFC] bump php to v8+ and Symfony to v5.4+
Hi

both php v7 and symfony v4 are EOL to date 
Imho it will be great to upgrade this bundle with such minimum requirements

wdyt ?

---

- Symfony v5.4 LTS ([release map](https://symfony.com/releases#symfony-releases-calendar))
  - https://github.com/meilisearch/meilisearch-symfony/pull/265
  - https://github.com/meilisearch/meilisearch-symfony/pull/301 (supersede the above)
- PHP v8 ([release map](https://www.php.net/supported-versions.php))
  - Blocked, see https://github.com/meilisearch/meilisearch-symfony/issues/263#issuecomment-1591498511

### PR DESCRIPTION
**Pull request by 94noni**
# Pull Request

## Related issue
Fixes #263 (Symfony part)

## What does this PR do?
- Upgrade to [Symfony v5.4 as LTS](https://symfony.com/releases#long-term-support-release) minimum (BC with EOL v4.4)
  - command definition via https://symfony.com/doc/5.4/console.html#creating-a-command
- Add also Symfony v6.3 in the CI process
- Leverage [Rector](https://github.com/rectorphp) to ease and automate upgrade (removed as per comments)

### CODE REVIEW

File: src/Command/MeilisearchClearCommand.php
Comment (norkunas) [inline src/Command/MeilisearchClearCommand.php:56]: Not sure if changes to constants add any value

Comment (94noni) [reply src/Command/MeilisearchClearCommand.php:56]: Its automated via rector, of course it is to be challenged
To me i am neither +1 nor -1

Comment (norkunas) [reply src/Command/MeilisearchClearCommand.php:56]: then it breaks CS, as the class should be imported not referenced via FQCN

Comment (94noni) [reply src/Command/MeilisearchClearCommand.php:56]: I did only run rector, not any cs fixer tool for now
Of course before i remove the WIP i will check this cs 👍🏻

**Comment by 94noni** [comment]
@norkunas should rector be kept (for later usage?)
PR is ready nonetheless

File: src/Command/MeilisearchImportCommand.php
Comment (brunoocasali) [inline src/Command/MeilisearchImportCommand.php:51]: Was it removed because it already contains the `default description`?

Comment (94noni) [reply src/Command/MeilisearchImportCommand.php:51]: It is related to https://github.com/rectorphp/rector-symfony/blob/main/docs/rector_rules_overview.md#commanddescriptiontopropertyrector

Comment (brunoocasali) [reply src/Command/MeilisearchImportCommand.php:51]: In our case we don't have it as a property, it will work the same way right?

**Comment by norkunas** [comment]
> @norkunas should rector be kept (for later usage?)

@brunoocasali should decide, but in case we need it very rarely if at all, maybe we shouldn't introduce as dev dependency to save some resources

**Comment by norkunas** [comment]
By the way Symfony 6.3 should be added to matrix

File: src/Command/MeilisearchImportCommand.php
Comment (94noni) [reply src/Command/MeilisearchImportCommand.php:51]: Yes but with deprecation triggedered
Or we should directly rely on the static prop
I am open to suggestion here

**Comment by 94noni** [comment]
Oki will do indeed the new 6.3
And remove rector for now as project is small so lets keep it simple

**Comment by brunoocasali** [comment]
> @brunoocasali should decide, but in case we need it very rarely if at all, maybe we shouldn't introduce as dev dependency to save some resources

I'm not opposed to keeping it, but my go-to opinion about this is: not to add new dependencies, even dev ones. So if we can remove it let's do it! :)

File: src/Command/MeilisearchImportCommand.php
Comment (94noni) [reply src/Command/MeilisearchImportCommand.php:51]: Copy that
Afk for 2 days but will come back to this shortly

**Comment by 94noni** [comment]
@brunoocasali @norkunas friendly ping, PR ready for reviews :)
cheers

**Comment by norkunas** [comment]
> @brunoocasali @norkunas friendly ping, PR ready for reviews :) cheers

lgtm

**Review by brunoocasali — DISMISSED** [review]
bors merge

**Comment by brunoocasali** [comment]
Can you check the CI error @94noni? 😇

**Comment by 94noni** [comment]
i will have a look at it indeed, hopeful soon :)

**Comment by 94noni** [comment]
i don't know why checks are not launched :s 
alos for reviewers, as this PR will need a major release, perhaps it could be great to also review https://github.com/meilisearch/meilisearch-symfony/pull/266 so we can remove such old services names prior to v1
thx

**Review by norkunas — APPROVED** [review]

**Comment by brunoocasali** [comment]
Hi @94noni, it is because you never got a PR merged. As soon you merge your first one, the CI will automatically run :) it is just a way to avoid spam (I think so)!

**Comment by norkunas** [comment]
6.1.* fails, I suggest to override `public static function getDefaultName()` and `public static function getDefaultDescription()` so they will work on all versions

**Comment by norkunas** [comment]
can you try `git commit --amend --no-edit && git push -f` to retrigger CI? not sure why `integration-tests (PHP 8.2) (Symfony 6.2.*)` does not run

**Comment by brunoocasali** [comment]
bors try

**Comment by norkunas** [comment]
We should merge #266 before

**Comment by brunoocasali** [comment]
Yeah, I was trying to force this CI action to trigger but had no success, either! Let's fix the conflicts then after approved we can see what is happen.

**Comment by 94noni** [comment]
as https://github.com/meilisearch/meilisearch-symfony/pull/266 will hopefully be merged shortly, I will rework on this right after
thx

**Comment by brunoocasali** [comment]
I just merged the #266 @94noni thanks a lot for your help :)

**Comment by brunoocasali** [comment]
bors merge

**Comment by curquiza** [comment]
Hello @94noni can you rebase your branch please? 😊 
There are git conflicts so we cannot merge

**Comment by 94noni** [comment]
@curquiza thx for the ping, I've tried resolving on GitHub.com ui, hope it is good
i will not have time available to finish soon this PR and thus also the following one https://github.com/meilisearch/meilisearch-symfony/pull/275

@norkunas please feel free to continue and proceed with PRs

File: src/Command/MeilisearchClearCommand.php
Comment (norkunas) [inline src/Command/MeilisearchClearCommand.php:25]: Revert this, as there is no point to do this at current time. When we'll be able to provide info via attributes we will.
Same in other commands

**Comment by Chris53897** [comment]
@94noni Thanks for your work. Do you have some time to finish this PR? After that PR is merged, Symfony 7 support will be easier to implement.

**Comment by 94noni** [comment]
@Chris53897 not so much sadly at the moment
If anyone wants to take over and finish it i will be more than happy to review etc

**Comment by 94noni** [comment]
closing in favor of https://github.com/meilisearch/meilisearch-symfony/pull/301 which takes over
thanks for original reviewers and @Chris53897 for continuing :)

**Pull request closed without merging**
